### PR TITLE
SMTChecker: Fix error that reports invalid number of verified checks

### PR DESCRIPTION
**Pull request by pgebal**
Solves https://github.com/ethereum/solidity/issues/15188 for BMC egine.

### CODE REVIEW

File: test/libsolidity/smtCheckerTests/overflow/signed_guard_sum_overflow.sol [hunk 1]
```diff
@@ -10,4 +10,4 @@ contract C  {
 // ----
 // Warning 3944: (78-83): CHC: Underflow (resulting value less than -2**255) happens here.
 // Warning 4984: (78-83): CHC: Overflow (resulting value larger than 2**255 - 1) happens here.
-// Info 1391: CHC: 1 verification condition(s) proved safe! Enable the model checker option "show proved safe" to see all of them.
+// Info 1391: CHC: 2 verification condition(s) proved safe! Enable the model checker option "show proved safe" to see all of them.
```
Comment (pgebal) [inline test/libsolidity/smtCheckerTests/overflow/signed_guard_sum_overflow.sol:13]: I think that comes from a fact that if the require statement is satisfied than there is no overflow or underflow in `x + y` expression.

File: libsolidity/formal/BMC.h [hunk 2]
```diff
@@ -149,7 +149,10 @@ class BMC: public SMTEncoder
 
 		friend bool operator<(BMCVerificationTarget const& _a, BMCVerificationTarget const& _b)
 		{
-			return _a.expression->id() < _b.expression->id();
+			if (_a.expression->id() == _b.expression->id())
```
Comment (pgebal) [inline libsolidity/formal/BMC.h:152]: Here was the problem. We didn't accumulate more than  1 proved BMC target for a given expression.

File: libsolidity/formal/BMC.cpp [hunk 3]
```diff
@@ -118,13 +118,18 @@ void BMC::analyze(SourceUnit const& _source, std::map<ASTNode const*, std::set<V
 		);
 
 	if (!m_settings.showProvedSafe && !m_safeTargets.empty())
+	{
+		std::size_t provedSafeNum = 0;
+		for (auto e: m_safeTargets)
+			provedSafeNum += e.second.size();
```
Comment (blishko) [inline libsolidity/formal/BMC.cpp]: In the current version, you would be making a copy of each entry in `m_safeTargets`, no?

```suggestion
		for (auto&& [_, targets]: m_safeTargets)
			provedSafeNum += targets.size();
```

File: libsolidity/formal/BMC.h (hunk 2 above)
Comment (blishko) [reply libsolidity/formal/BMC.h:152]: Right, good catch!

**Review by blishko — CHANGES_REQUESTED** [review]
Let's not make unnecessary copies.
Otherwise, this is good!

**Review by blishko — APPROVED** [review]
LGTM!

**Pull request merged**